whois-query/src/main/java/net/ripe/db/whois/query/pipeline/QueryDecoder.java

```diff
@@ -5,20 +5,33 @@
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.MessageToMessageDecoder;
 import net.ripe.db.whois.common.Message;
+import net.ripe.db.whois.common.QueryMessage;
+import net.ripe.db.whois.query.QueryFlag;
+import net.ripe.db.whois.query.QueryMessages;
+import net.ripe.db.whois.query.QueryParser;
 import net.ripe.db.whois.query.acl.AccessControlListManager;
+import net.ripe.db.whois.query.domain.QueryCompletionInfo;
+import net.ripe.db.whois.query.domain.QueryException;
 import net.ripe.db.whois.query.query.Query;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 import java.net.InetSocketAddress;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.charset.UnsupportedCharsetException;
 import java.util.List;
 
+import static net.ripe.db.whois.query.pipeline.WhoisEncoder.CHARSET_ATTRIBUTE;
+
 @Component
 @ChannelHandler.Sharable
 public class QueryDecoder extends MessageToMessageDecoder<String> {
 
     private final AccessControlListManager accessControlListManager;
 
+    private static final Charset DEFAULT_CHARSET = StandardCharsets.ISO_8859_1;
```

Comment:
Move all this logic to the `Query` class to remove duplication, i.e use `query.getCharset()` which returns user-specified charset or latin-1 by default.

Reply:
I did it but called getDefaultCharset instead